server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -135,30 +194,91 @@ export class TenantManager {
     }
 
     /**
-     * Retrieves the raw databasse tenant document
+     * Generates a new key for a tenant
      */
-    private async getTenantDocument(tenantId: string): Promise<ITenantDocument> {
+    public async refreshTenantKey(tenantId: string): Promise<string> {
         const db = await this.mongoManager.getDatabase();
         const collection = db.collection<ITenantDocument>(this.collectionName);
 
-        const found = await collection.findOne({ _id: tenantId });
+        const tenantKey = this.generateTenantKey();
+        const encryptedTenantKey = this.secretManager.encryptSecret(tenantKey);
+        if (encryptedTenantKey == null) {
+            winston.error("Tenant key encryption failed.");
+            return Promise.reject("Tenant key encryption failed.");
+        }
 
+        await collection.update({ _id: tenantId }, { encryptedTenantKey }, null);
+
+        return tenantKey;
+    }
+
+    /**
+     * Attaches fields to older tenants to provide backwards compatibility
+     */
+    private attachDefaultsToTenantDocument(tenantDocument: ITenantDocument): void {
```

Comment:
Why we need back compat? Shouldn't all tenants have storage and orderer anyway? And custom data can be undefined?

Reply:
This was previously existing functionality. I moved it to a function so it could be used within both `getTenantDocument` and `getAllTenantDocuments`

Reply:
Ah I see. I guess we did not have the fields for very long time. We should leave a comment and say it will removed at some point.